src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/DfaMatchingState.cs

```diff
@@ -118,24 +118,23 @@ internal DfaMatchingState<T> Next(T minterm)
         /// </summary>
         /// <param name="minterm">minterm corresponding to some input character or False corresponding to last \n</param>
         /// <returns>an enumeration of the transitions as pairs of the target state and a list of effects to be applied</returns>
-        internal List<(DfaMatchingState<T> derivative, List<DerivativeEffect> effects)> NfaEagerNextWithEffects(T minterm)
+        internal List<(DfaMatchingState<T> derivative, DerivativeEffect[] effects)> NfaEagerNextWithEffects(T minterm)
```

Comment:
What does the "eager" naming mean in these contexts?  Could you add that to the XML comments somewhere?